internal/cmd/grpc.go

```diff
@@ -228,11 +229,12 @@ func NewGRPCServer(
 		})),
 		grpc_prometheus.UnaryServerInterceptor,
 		otelgrpc.UnaryServerInterceptor(),
+		middlewaregrpc.ErrorUnaryInterceptor,
```

Comment:
this ErrorInterceptor needs to come before authz/authn as the errors returned from those middleware should be caught by the error interceptor and interceptors work from outside in, so the first ones are on the outer layers